### PR TITLE
Readme files

### PR DESCRIPTION
**Pull request by mpinson-NTIA**
The VQM software repository is currently described on the ITS website. This pull request moves that information into the repository. Some information and links have been updated.

### CODE REVIEW

File: CVQM.md
Comment (lsegreNTIA) [inline CVQM.md:1]: Please capitalize NTIA TR

File: CVQM.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Technical details are available in [TR-02-392](https://its.ntia.gov/publications/details?pub=2423) and [TR-06-433b](https://its.ntia.gov/publications/details?pub=2465).
+
+### 'none'
+No calibration will be performed. Assume that the first frame in original and processed video files align temporally. Run model with default calibration values which assumes that the processed video file is perfectly calibrated.
+
+### 'manual' 
+Read the calibration file created on a previous run. The values on the beginning of each line may be manually modified.
+
+### 'rrcal' 
+Perform reduced reference calibration as given in ntia_tr_06_433a.pdf (except assume no spatial scaling). These algorithms use random processes, which may yield slightly different results from one run to another.
```
Comment (lsegreNTIA) [inline CVQM.md]: Please capitalize NTIA TR and remove the .pdf. Really, a properly formatted hyperlink as above would be preferable.

File: CVQM.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+### 'manual' 
+Read the calibration file created on a previous run. The values on the beginning of each line may be manually modified.
+
+### 'rrcal' 
+Perform reduced reference calibration as given in ntia_tr_06_433a.pdf (except assume no spatial scaling). These algorithms use random processes, which may yield slightly different results from one run to another. 
+It is highly recommended that rrcal results be median filtered across 7 or more different video sequences that have been sent through the same video system (see Calibration Note below).
+
+### 'rrcalscale' 
+Perform reduced reference calibration as given in ntia_tr_06_433a.pdf, including estimating spatial scaling (e.g., stretch). These algorithms use random processes, which may yield slightly different results from one run to another.
```
Comment (lsegreNTIA) [inline CVQM.md]: NTIA TR -- see line 34 comment

File: CVQM.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+* 'developers' - Developer's model, a fast running variant of the General Model, described in [TR-02-392](https://its.ntia.gov/publications/details?pub=2423)
+* 'lowbw' - Low Bandwidth Model, described [here](https://its.ntia.gov/publications/details?pub=2575)
+* 'fastlowbw' - Fast Low Bandwidth Model, a fast running variant of the low bandwidth model 
+
+The CVQM software provides a clean implementation of these VQM models. 
+It is intended to be ported into other programming languages or to be called from other software. 
+
+## Calibration Options
+The following calibration options are available. 
+Technical details are available in [TR-02-392](https://its.ntia.gov/publications/details?pub=2423) and [TR-06-433b](https://its.ntia.gov/publications/details?pub=2465).
```
Comment (lsegreNTIA) [inline CVQM.md]: NTIA TR is now preferred.

File: CVQM.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+### 'rrcal' 
+Perform reduced reference calibration as given in ntia_tr_06_433a.pdf (except assume no spatial scaling). These algorithms use random processes, which may yield slightly different results from one run to another. 
+It is highly recommended that rrcal results be median filtered across 7 or more different video sequences that have been sent through the same video system (see Calibration Note below).
+
+### 'rrcalscale' 
+Perform reduced reference calibration as given in ntia_tr_06_433a.pdf, including estimating spatial scaling (e.g., stretch). These algorithms use random processes, which may yield slightly different results from one run to another. 
+It is highly recommended that rrcalscale results be median filtered across 7 or more different video sequences that have been sent through the same video system (see Calibration Note below).
+
+### 'rrcal2' 
+Improved version of 'rrcal' - version 2, as specified in ITU-T Recommendation J.244 and ntia_tr_08_433b.pdf (see also 'rrcal2scale'). Includes estimate of Cb and Cr gain and offset; and slightly improved luminance gain & offset algorithm.
```
Comment (lsegreNTIA) [inline CVQM.md]: See line 34 comment

File: CVQM.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+### 'rrcal2scale' 
+Improved version of 'rrcalscale' - version 2, as specified in ITU-T Recommendation J.244 and ntia_tr_08_433b.pdf (see also 'rrcal2'). Includes estimate of Cb and Cr gain and offset; and slightly improved luminance gain & offset algorithm.
```
Comment (lsegreNTIA) [inline CVQM.md]: ditto

File: README.md [hunk 6]
```diff
@@ -1,8 +1,11 @@
...
+This repository contains the Video Quality Metric (VQM) software developed by ITS between 2002 and 2010. 
+The VQM algorithms estimate how people perceive video quality. These VQM algorithms compare the processed video (output) with the original video (input). These models are suitable when the original video is good quality or better. Quality problems from the camera and original production are not considered. 
+The VQM tools are available free of charge for any commercial or non-commercial use, in accordance with the terms described in the [LICENSE](https://github.com/NTIA/vqm/blob/master/LICENSE.md).
 
 For each software package, there are three download options: the MATLAB® source, a 32-bit compiled version, and a 64-bit compiled version. In each case, the relevant documentation is provided. 
+These software packages are no longer supported.
 
 The available software packages are:
```
Comment (lsegreNTIA) [inline README.md:9]: Delete?

File: README.md [hunk 7]
```diff
@@ -26,12 +29,65 @@ Each metric is stored in this repository such that for the given version of a gi
 |- <nmae>_pc64_v123     # directory contains the 64-bit executable for that metric version.
```
Comment (lsegreNTIA) [inline README.md]: name

File: README.md [hunk 8]
```diff
@@ -26,12 +29,65 @@ Each metric is stored in this repository such that for the given version of a gi
...
-
 Readmes on how to use each metric are available under the name `<name>_pc_readme.pdf`.
 
 Note that this distribution does not include the MATLAB Runtime, which must first be installed before running any of the compiled versions of these metrics. The runtime can be installed using `MCRInstaller.exe`, located [here](https://www.mathworks.com/products/compiler/matlab-runtime.html). Be aware that the compiled versions of the metrics were compiled in MATLAB R2013b (8.2). Future versions of MATLAB may cause the code not to run. This issue applies to both the compiled code and the source code. 
 
+## Usage
+
+### Batch Video Quality Metric (BVQM) Software 
+BVQM is a Windows program for performing out-of-service / lab bench testing. 
+BVQM can processing and analyses of multiple video scenes and multiple video systems at once.
```
Comment (lsegreNTIA) [inline README.md]: can perform processing and

**Review by lsegreNTIA — APPROVED** [review]
Approve conditioned on corrections noted

**Review by lsegreNTIA — APPROVED** [review]
Approve conditioned on corrections noted

File: CVQM.md (hunk 1 above)
Comment (mpinson-NTIA) [reply CVQM.md]: I corrected all of these references to TR-06-433 and updated them from 433a to 433b.

File: CVQM.md (hunk 2 above)
Comment (mpinson-NTIA) [reply CVQM.md]: Done.

File: CVQM.md (hunk 3 above)
Comment (mpinson-NTIA) [reply CVQM.md]: Fixed throughout.

File: CVQM.md (hunk 4 above)
Comment (mpinson-NTIA) [reply CVQM.md]: done

File: CVQM.md (hunk 5 above)
Comment (mpinson-NTIA) [reply CVQM.md]: done

File: README.md (hunk 6 above)
Comment (mpinson-NTIA) [reply README.md:9]: sure

File: README.md (hunk 7 above)
Comment (mpinson-NTIA) [reply README.md]: fixed

File: README.md (hunk 8 above)
Comment (mpinson-NTIA) [reply README.md]: fixed

File: CVQM.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+Every original / processed video sequence pair is run through two main steps. First, the requested calibration is run, and results are saved in a file named after the processed video file, with '_calibration.txt' appended. Second, the requested model is run, and results are saved in a file named after the processed video file, with '_model.txt' appended.
+
+Any errors that occur are recorded in a file named after the processed video file, with '_errors.txt' appended. If that file is absent, no errors occurred. Each line on this file will start with a number, indicating error type, and then a string describing the error. 
+Type 1 errors are fatal data input/output issues (e.g., operation cannot continue due to invalid input argument, file read error, or file write error). Type 2 errors are calibration warnings (e.g., still sequence, temporal registration failure, spatial shift beyond search limits). Type 3 errors are non-fatal warnings.
+
+## VQM Model Options
+The following VQM models options are available:
```
Comment (blainNTIA) [inline CVQM.md]: Is "... models options ..." correct?
Should this be "model options" or "model's options"?

File: CVQM.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+* 'none'- No model will be calculated - run calibration only (see below).
+* 'general' - NTIA General Model, as given in the ANSI T1.801.03-2003, ITU-T J.144 (03/04), ITU-R BT.1683 (06/04), and [NTIA TR-02-392](https://its.ntia.gov/publications/details?pub=2423). 
+* 'developers' - Developer's model, a fast running variant of the General Model, described in [NTIA TR-02-392](https://its.ntia.gov/publications/details?pub=2423)
+* 'lowbw' - Low Bandwidth Model, described [here](https://its.ntia.gov/publications/details?pub=2575)
```
Comment (blainNTIA) [inline CVQM.md]: Using "here" as the link word seems a bit disjoint when all the others are like "NTIA TR-02-392". Maybe use the paper's title like shown below?
```suggestion
* 'lowbw' - Low Bandwidth Model, described in the paper [Low Bandwidth Reduced Reference Video Quality Monitoring System](https://its.ntia.gov/publications/details?pub=2575)
```

File: README.md [hunk 11]
```diff
@@ -16,22 +18,75 @@ The available software packages are:
...
+The In-Service Video Qualtiy Metric (IVQM) was a "proof of concept" software implementation.
+This software demonstrates that the reduced reference (RR) algorithms provided within BVQM and CVQM can actually run in-service. 
+
+The IVQM software is no longer available. 
+The video capture code is not compatible with any current video capture device.
+The remainder of the code is provided by the CVQM software package. 
+
+### VQM - Obsolete - Not Available
+VQM_pc (compiled for Windows) and VQM_lx (compiled for Linux) have been replaced by BVQM. 
+This software was developped under a cooperative research and devleopment agreement (CRADA) by ITS and Intel.
```
Comment (blainNTIA) [inline README.md]:
```suggestion
This software was developed under a cooperative research and development agreement (CRADA) by ITS and Intel. 
```

File: README.md [hunk 12]
```diff
@@ -16,22 +18,75 @@ The available software packages are:
...
+### VFD
+This command-line MATLAB software calculates variable frame delay (VFD) between an original video sequence and a processed video sequence. This algorithm and code is used by BVQM, CVQM, and PSNR software packages.
+The algorithm is described [here](https://its.ntia.gov/publications/2500.aspx).
+
+### FDF
+This command-line MATLAB software calculates fraction dropped frames (FDF) from VFD statistics. This algorithm and code is used by BVQM, CVQM, and PSNR software packages.
+The algorithm is described [here](https://its.ntia.gov/publications/2493.aspx).
+
+### IVQM - Obsolete - Not Available
+The In-Service Video Qualtiy Metric (IVQM) was a "proof of concept" software implementation.
```
Comment (blainNTIA) [inline README.md]:
```suggestion
The In-Service Video Quality Metric (IVQM) was a "proof of concept" software implementation.
```

File: README.md [hunk 13]
```diff
@@ -16,22 +18,75 @@ The available software packages are:
...
+### PSNR
+This command-line MATALAB software calculates peak signal to noise ratio (PSNR) according to ITU-T Rec. J.340. Also computes PSNR with variable frame delays removed (PSNR-VFD). These metrics are also available in the BVQM and CVQM software.
+The algorithm is described [here](https://its.ntia.gov/publications/details.aspx?pub=2500).
+
+### VFD
+This command-line MATLAB software calculates variable frame delay (VFD) between an original video sequence and a processed video sequence. This algorithm and code is used by BVQM, CVQM, and PSNR software packages.
+The algorithm is described [here](https://its.ntia.gov/publications/2500.aspx).
+
+### FDF
+This command-line MATLAB software calculates fraction dropped frames (FDF) from VFD statistics. This algorithm and code is used by BVQM, CVQM, and PSNR software packages.
```
Comment (blainNTIA) [inline README.md]:
```suggestion
This command-line MATLAB software calculates fraction dropped frames (FDF) from VFD statistics. This algorithm and code are used by BVQM, CVQM, and PSNR software packages.
```

File: README.md [hunk 14]
```diff
@@ -16,22 +18,75 @@ The available software packages are:
...
+The two differences are that (1) CVQM is called from a command line (e.g., Windows Accessory "Command Prompt"), and (2) CVQM cannot compare the results from multiple video clips to improve calibration accuracy.  
+CVQM runs on one pair of video files at a time, and writes results to files. 
+
+* CVQM_pc32_v30 has CVQM version 3.0 compiled for the 32-bit Windows operating system.
+* CVQM_pc64_v30 has CVQM version 3.0 compiled for the 64-bit Windows operating system.
+* CVQM_src_v30 has MATLAB source code for CVQM version 3.0.\
+* [Documentation](CVQM.md)
+
+CVQM is a clean implementation of the VQM metrics that are available in BVQM. 
+The software functionality is split to support an in-service RR system (i.e., live calculations  on the upstream and downstream video feeds, communicating via a secondary communication channel).
```
Comment (blainNTIA) [inline README.md]: Two spaces in "... calculations  on..."
```suggestion
The software functionality is split to support an in-service RR system (i.e., live calculations on the upstream and downstream video feeds, communicating via a secondary communication channel). 
```

File: README.md [hunk 15]
```diff
@@ -16,22 +18,75 @@ The available software packages are:
...
+
+* BVQM_pc32_v20 has BVQM version 2.0, compiled for 32-bit Windows operating systems.
+* BVQM_pc64_v20 has BVQM version 2.0, compiled for 64-bit Windows operating systems.
+* BVQM_src_v20 has MATLAB source code for BVQM 2.0.
+* [Documentation](https://its.ntia.gov/publications/details.aspx?pub=2558)
+
+BVQM contains unnecessary functionality and residual code remaining from the metric development process. 
+
+### Command Line Video Quality Metric (CVQM) Software
+CVQM is a Windows command line program for performing out-of-service / lab bench testing. The video quality calibration and metric algorithms within CVQM are basically identically to those offered by BVQM.
```
Comment (blainNTIA) [inline README.md]:
```suggestion
CVQM is a Windows command line program for performing out-of-service / lab bench testing. The video quality calibration and metric algorithms within CVQM are basically identical to those offered by BVQM. 
```

File: README.md [hunk 16]
```diff
@@ -16,22 +18,75 @@ The available software packages are:
...
+
+CVQM is a clean implementation of the VQM metrics that are available in BVQM. 
+The software functionality is split to support an in-service RR system (i.e., live calculations  on the upstream and downstream video feeds, communicating via a secondary communication channel). 
+
+### PSNR
+This command-line MATALAB software calculates peak signal to noise ratio (PSNR) according to ITU-T Rec. J.340. Also computes PSNR with variable frame delays removed (PSNR-VFD). These metrics are also available in the BVQM and CVQM software.
+The algorithm is described [here](https://its.ntia.gov/publications/details.aspx?pub=2500).
+
+### VFD
+This command-line MATLAB software calculates variable frame delay (VFD) between an original video sequence and a processed video sequence. This algorithm and code is used by BVQM, CVQM, and PSNR software packages.
```
Comment (blainNTIA) [inline README.md]:
```suggestion
This command-line MATLAB software calculates variable frame delay (VFD) between an original video sequence and a processed video sequence. This algorithm and code are used by BVQM, CVQM, and PSNR software packages.
```

**Review by blainNTIA — APPROVED** [review]
Overall, it looks great. I made some suggestions to call out possible typos. Please review them. Overall, it looks great.

File: CVQM.md (hunk 9 above)
Comment (mpinson-NTIA) [reply CVQM.md]: changed to VQM software's options

File: CVQM.md (hunk 10 above)
Comment (mpinson-NTIA) [reply CVQM.md]: done.

File: README.md (hunk 11 above)
Comment (mpinson-NTIA) [reply README.md]: fixed

File: README.md (hunk 12 above)
Comment (mpinson-NTIA) [reply README.md]: fixed.

File: README.md (hunk 13 above)
Comment (mpinson-NTIA) [reply README.md]: done.

File: README.md (hunk 14 above)
Comment (mpinson-NTIA) [reply README.md]: done

File: README.md (hunk 15 above)
Comment (mpinson-NTIA) [reply README.md]: done

File: README.md (hunk 16 above)
Comment (mpinson-NTIA) [reply README.md]: done

**Review by jkubNTIA — APPROVED** [review]

**Comment by jkubNTIA** [comment]
I reviewed for Angela as a delegate supervisor since Angela does not have an account on Github.

**Pull request merged**